### ISSUE TITLE
Model attribute beginning with "_" leads to a NULL value

### ISSUE THREAD

**Issue by segalaj** [wisdom-framework/wisdom-orientdb#23]
**Problem**
Get the value of an element attribute named *_\<something\>* return always NULL

**How to reproduce**
Create a model like:
```
@Entity
public class Test {
    @Id
    private String id;

    private Date _date;

    public Test() {
    }

    public String getId() {
        return id;
    }

    public Date getDate() {
        return _date;
    }

    public void setDate(Date date) {
        this._date = date;
    }
}
```
Try to access to an element in a controller:
```
....
    @Model(value = Test.class)
    private OrientDbCrud<Test,String> _testCrud;
....
    @Route(method = HttpMethod.GET, uri = "")
    public Result test() {
        return ok(Iterables.toArray(_testCrud.findAll(), License.class)).json();
    }
```

Return:
> [{"id":"#18:0","date":null}]

**Environment**
Java 8, wisdom 0.7.0, wisdom-orientdb 0.2.4, orientdb 2.0.2

**Comment by barjo** [comment]
Hi,

I can reproduce it on the snapshot version. It seems that it's a limitation of orientdb itself (or possibly javassist). We should consider opening the issue or their tracker https://github.com/orientechnologies/orientdb/issues

**Comment by segalaj** [comment]
Hi, thanks for your answer.
I'm not sure that come from oriendb, in fact it's possible to have column name beginning with "_" in one table.

What do you think?

**Comment by cescoffier** [comment]
Hi,

It may come from the proxy generation that orientDB is doing.

2015-03-26 13:38 GMT+01:00 Joachim Segala <notifications@github.com>:

**Comment by barjo** [comment]
Indeed, I am pretty sure it's related to the JPA layer of orientdb. I will check with them and open an issue if that's the case.

**Comment by barjo** [comment]
After further investigation it seems that it is because the field name does not match the getter/setter. 
https://github.com/orientechnologies/orientdb/blob/master/object/src/main/java/com/orientechnologies/orient/object/enhancement/OObjectMethodFilter.java

**Comment by barjo** [comment]
After more more further investigation; The problem appears only when we don't use OObjectDatabaseTx to create the object instance. For example, if we use 
Test test = db.newInstance(Test.class) to create the instance, we don't have the problem. It seems that the problem appears with the proxy creation in the case the object was not created by new Instance.

**Comment by barjo** [comment]
Actually it doesn't work too, the saved return instance will contains the field but it won't actually be persisted.

**Comment by barjo** [comment]
I open an issue at orientdb https://github.com/orientechnologies/orientdb/issues/3822

**Comment by segalaj** [comment]
Thanks a lot for your investigation.

**Comment by barjo** [comment]
It seems it will be fix for orientdb 2.2
